skrifa/src/outline/autohint/axis.rs

```diff
@@ -22,23 +27,43 @@ pub struct Axis {
     pub segments: SmallVec<Segment, MAX_INLINE_SEGMENTS>,
 }
 
-/// Either horizontal or vertical.
-///
-/// A type alias because it's used as an index.
-pub type Dimension = usize;
-
 impl Axis {
     /// X coordinates, i.e. vertical segments and edges.
     pub const HORIZONTAL: Dimension = 0;
     /// Y coordinates, i.e. horizontal segments and edges.
     pub const VERTICAL: Dimension = 1;
 }
 
+impl Axis {
+    pub fn new(dim: Dimension, orientation: Option<Orientation>) -> Self {
+        let mut axis = Self::default();
+        axis.reset(dim, orientation);
+        axis
+    }
+
+    pub fn reset(&mut self, dim: Dimension, orientation: Option<Orientation>) {
+        self.dim = dim;
+        if dim == Self::HORIZONTAL {
+            self.major_dir = Direction::Up;
+        } else {
+            self.major_dir = Direction::Left;
+        }
+        if orientation == Some(Orientation::Clockwise) {
+            if dim == Self::HORIZONTAL {
+                self.major_dir = Direction::Down;
+            } else {
+                self.major_dir = Direction::Right;
+            }
+        }
```

Comment:
totally academic but I'd be tempted to write this as, `let self.major_dir = match (dim, orientation) { .. }`

Reply:
I didn't even try a match because of the multiple catch-all conditions but it does look much better. Fixed